promlog/flag/flag.go

```diff
@@ -25,9 +25,19 @@ const LevelFlagName = "log.level"
 // LevelFlagHelp is the help description for the log.level flag.
 const LevelFlagHelp = "Only log messages with the given severity or above. One of: [debug, info, warn, error]"
 
+// FormatFlagName is the canonical flag name to configure the log format
+// within Prometheus projects.
+const FormatFlagName = "log.format"
+
+// FormatFlagHelp is the help description for the log.format flag.
+const FormatFlagHelp = "Output format of log messages. One of: [logfmt, json]"
+
 // AddFlags adds the flags used by this package to the Kingpin application.
 // To use the default Kingpin application, call AddFlags(kingpin.CommandLine)
 func AddFlags(a *kingpin.Application, logLevel *promlog.AllowedLevel) {
 	a.Flag(LevelFlagName, LevelFlagHelp).
 		Default("info").SetValue(logLevel)
+
+	a.Flag(FormatFlagName, FormatFlagHelp).
+		Default("logfmt").Enum("logfmt", "json")
```

Comment:
I might be missing something obvious but I don't see how this is going to work for Prometheus. How will the `main()` know about the flag's value?

Reply:
It shouldn't need to.